PCbuild/get_external.py

```diff
@@ -53,7 +55,24 @@ def main():
         verbose=args.verbose,
     )
     final_name = args.externals_dir / args.tag
-    extract_zip(args.externals_dir, zip_path).replace(final_name)
+    extracted = extract_zip(args.externals_dir, zip_path)
+    for retries in range(5, 0, -1):
+        try:
+            extracted.replace(final_name)
+            break
+        except PermissionError as ex:
+            if retries > 1:
+                print(f"Encountered permission error '{ex}'. Retrying in five seconds...", file=sys.stderr)
+                time.sleep(5)
+            else:
+                print(f"Encountered permission error '{ex}'", file=sys.stderr)
```

Comment:
Not that it matters in this particular case but I like to use exponential backoff when doing retries. This also happens to make the logic cleaner:

```suggestion
    for wait in [1, 2, 3, 5, 8, 0]:
        try:
            extracted.replace(final_name)
            break
        except PermissionError as ex:
            retry = f" Retrying in {wait}s..." if wait else ""
            print(f"Encountered permission error '{ex}'.{retry}", file=sys.stderr)
            time.sleep(wait)
```